[Bug]: Compile warnings with release

### Version of OpenTTD

master

### Expected result

No compile warnings :D

### Actual result

`OpenTTD/src/engine.cpp:541:7: warning: variable 'index' set but not used [-Wunused-but-set-variable]`
`OpenTTD/src/3rdparty/squirrel/squirrel/sqstate.cpp:453:20: warning: variable 'nfound' set but not used [-Wunused-but-set-variable]`

Both related to release builds, which build without `assert`. Both seem legit.

### Steps to reproduce

Build with `-DCMAKE_BUILD_TYPE=Release -DOPTION_USE_ASSERTS=OFF`

--- a/cmake/CompileFlags.cmake
+++ b/cmake/CompileFlags.cmake
@@ -56,7 +56,7 @@ macro(compile_flags)
 
     if(MSVC)
         add_compile_options(/W3)
-        if(MSVC_VERSION GREATER 1929)
+        if(MSVC_VERSION GREATER 1929 AND CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
             # Starting with version 19.30, there is an optimisation bug, see #9966 for details
             # This flag disables the broken optimisation to work around the bug
             add_compile_options(/d2ssa-rse-)

--- a/src/3rdparty/squirrel/squirrel/sqstate.cpp
+++ b/src/3rdparty/squirrel/squirrel/sqstate.cpp
@@ -450,7 +450,7 @@ void RefTable::Resize(SQUnsignedInteger size)
 	SQUnsignedInteger oldnumofslots = _numofslots;
 	AllocNodes(size);
 	//rehash
-	SQUnsignedInteger nfound = 0;
+	[[maybe_unused]] SQUnsignedInteger nfound = 0;
 	for(SQUnsignedInteger n = 0; n < oldnumofslots; n++) {
 		if(type(t->obj) != OT_NULL) {
 			//add back;

--- a/src/engine.cpp
+++ b/src/engine.cpp
@@ -538,7 +538,7 @@ void SetupEngines()
 	_engine_pool.CleanPool();
 
 	assert(_engine_mngr.size() >= _engine_mngr.NUM_DEFAULT_ENGINES);
-	uint index = 0;
+	[[maybe_unused]] uint index = 0;
 	for (const EngineIDMapping &eid : _engine_mngr) {
 		/* Assert is safe; there won't be more than 256 original vehicles
 		 * in any case, and we just cleaned the pool. */

